bfcfg

```diff
@@ -48,54 +48,130 @@ log_msg()
   echo "$@" >> ${log_file}
 }
 
+mfg_lock()
+{
+  log_msg "mfg: lock the partition"
+  echo 1 > /sys/bus/platform/drivers/mlx-bootctl/mfg_lock 2>>${log_file}
+}
+
 mfg_cfg()
 {
   local tmp_file=/tmp/.bfcfg-mfg-data
-  local opn_str_sysfs=/sys/bus/platform/drivers/mlx-bootctl/opn_str
+  local opn_sysfs=/sys/bus/platform/drivers/mlx-bootctl/opn
+  local sku_sysfs=/sys/bus/platform/drivers/mlx-bootctl/sku
+  local modl_sysfs=/sys/bus/platform/drivers/mlx-bootctl/modl
+  local sn_sysfs=/sys/bus/platform/drivers/mlx-bootctl/sn
+  local uuid_sysfs=/sys/bus/platform/drivers/mlx-bootctl/uuid
   local oob_efi_mac_sysfs=${efivars}/OobMacAddr-${efi_global_var_guid}
   local mac_err opn_err mac
 
   if [ $dump_mode -eq 1 ]; then
     [ -e "${oob_mac_sysfs}" ] && echo "mfg: MFG_OOB_MAC=$(cat ${oob_mac_sysfs} 2>/dev/null)"
-    [ -e "${opn_str_sysfs}" ] && echo "mfg: MFG_OPN_STR=$(cat ${opn_str_sysfs} 2>/dev/null)"
+    [ -e "${opn_sysfs}" ] && echo "mfg: MFG_OPN=$(cat ${opn_sysfs} 2>/dev/null)"
+    [ -e "${sku_sysfs}" ] && echo "mfg: MFG_SKU=$(cat ${sku_sysfs} 2>/dev/null)"
+    [ -e "${modl_sysfs}" ] && echo "mfg: MFG_MODL=$(cat ${modl_sysfs} 2>/dev/null)"
+    [ -e "${sn_sysfs}" ] && echo "mfg: MFG_SN=$(cat ${sn_sysfs} 2>/dev/null)"
+    [ -e "${uuid_sysfs}" ] && echo "mfg: MFG_UUID=$(cat ${uuid_sysfs} 2>/dev/null)"
     return
   fi
 
-  if [ -n "${MFG_OOB_MAC}" -a -z "${MFG_OPN_STR}" ] ||
-     [ -z "${MFG_OOB_MAC}" -a -n "${MFG_OPN_STR}" ]; then
-    ${ECHO} "ERROR: both MFG_OOB_MAC and MFG_OPN_STR must be specified"
-    ${ECHO} "       in order to write information to the MFG partition."
+  if [ -z "${MFG_OOB_MAC}" ] || [ -z "${MFG_OPN}" ] || [ -z "${MFG_SKU}" ] ||
+     [ -z "${MFG_MODL}" ] || [ -z "${MFG_SN}" ] || [ -z "${MFG_UUID}" ]; then
+    log_msg "mfg: skip mfg since one or more of the following are unspecified:"
+    log_msg "     MFG_OOB_MAC, MFG_OPN, MFG_SKU, MFG_MODL, MFG_SN, MFG_UUID"
```

Comment:
Once the changes are in, we'll need to notify the production team about these new options, or else their production flow would fail.

Reply:
Yes, I will sync with them on this